src/SFA.DAS.EmployerCommitmentsV2.Web/Controllers/CreateCohortController.cs

```diff
@@ -73,21 +80,26 @@ public async Task<IActionResult> SelectProvider(SelectProviderViewModel request)
                 }
 
                 providerResponse = await _commitmentsApiClient.GetProvider(long.Parse(request.ProviderId));
-                
+
                 var confirmProviderRequest = _confirmProviderRequestMapper.Map(request);
 
                 return RedirectToAction("ConfirmProvider", confirmProviderRequest);
             }
-            catch (RestHttpClientException e)
+            catch (RestHttpClientException ex)
             {
-                if (e.StatusCode == HttpStatusCode.NotFound)
+                if (ex.StatusCode == HttpStatusCode.NotFound)
                 {
                     ModelState.AddModelError(nameof(providerResponse.ProviderId), "Check UK Provider Reference Number");
                     return View(request);
                 }
+                    _logger.LogError($"Failed '{nameof(CreateCohortController)}-{nameof(SelectProvider)}': {nameof(ex.StatusCode)}='{ex.StatusCode}', {nameof(ex.ReasonPhrase)}='{ex.ReasonPhrase}'");
+            }
+            catch (Exception ex)
+            {
+                _logger.LogError($"Failed '{nameof(CreateCohortController)}-{nameof(SelectProvider)}': {nameof(ex.Message)}='{ex.Message}', {nameof(ex.StackTrace)}='{ex.StackTrace}'");
             }
 
-            return View(request);
+            return View("~/Views/Error/Error.cshtml");
```

Comment:
use RedirectToAction("Error", "Error") instead. in your test you can verify the ActionName and ControllerName.